### DISCUSSION TITLE
Usage with kitty as external terminal

### DISCUSSION THREAD

**Discussion by smackesey** [mfussenegger/nvim-dap#464] [Q&A]
I was going to post this as an issue (1) not sure if this is a kitty or an `nvim-dap` issue; and (2) getting a clean repro to fit the issue template will take some time. So trying here first.

Has anyone succeeded in getting `nvim-dap` to work with `kitty` as external terminal?

```
dap.defaults.fallback.external_terminal = {
  command = '/usr/local/bin/kitty',
  args = {
    '--hold',
  },
}
```

With this configuration, everything works _except_ that the kitty window closes when the debug process terminates. Kitty's `--hold` is supposed to keep the window open until the user manually quits. I was able to capture this error message printed immediately before the window closes:

<img width="345" alt="image" src="https://user-images.githubusercontent.com/1531373/157303103-5b7ecad3-6aa3-470b-b77a-920f928309d1.png">

So it looks like dap is sending some termination sequence when the session ends that kitty chokes on. Anyone know of a workaround?

**Comment by mfussenegger** [comment]
For what's worth the `hold` option for alacritty works.

```
  dap.defaults.fallback.external_terminal = {
    command = '/usr/bin/alacritty';
    args = {'--hold', '-e'};
  }
```

So I suspect this may be kitty related.

**Comment by chunleng** [comment]
Posting here a plugin I have created as an alternative:
https://github.com/chunleng/nvim-dap-kitty-launcher

The plugin currently supports using kitty's remote control to launch dap.
Currently it only supports python but I will add more upon user's request
